tests/tracing/test_misc.py

```diff
@@ -60,6 +61,37 @@ def test_transaction_naming(sentry_init, capture_events):
     assert events[2]["transaction"] == "a"
 
 
+def test_transaction_data(sentry_init, capture_events):
+    sentry_init(traces_sample_rate=1.0)
+    events = capture_events()
+
+    with start_transaction(name="test-transaction"):
+        span_or_tx = sentry_sdk.get_current_span()
+        span_or_tx.set_data("foo", "bar")
+        with start_span(op="test-span") as span:
+            span.set_data("spanfoo", "spanbar")
+
+    transaction = events[0]
```

Comment:
Do we also want to assert here whether there is exactly one event? With how the test is currently written, the test would pass if there are multiple events.

I am guessing we probably want something like the following:

```
(transaction,) = events
```